packages/shared/src/components/onboarding/OnboardingFeedHeader.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useCallback, useEffect } from 'react';
+import classNames from 'classnames';
+import { useRouter } from 'next/router';
+import { FilterOnboardingV4 } from './FilterOnboardingV4';
+import {
+  Button,
+  ButtonColor,
+  ButtonIconPosition,
+  ButtonSize,
+  ButtonVariant,
+} from '../buttons/Button';
+import { REQUIRED_TAGS_THRESHOLD } from './common';
+import { ArrowIcon } from '../icons';
+import { LayoutHeader } from '../layout/common';
+import { usePrompt } from '../../hooks/usePrompt';
+import { useMyFeed } from '../../hooks/useMyFeed';
+import { useAlertsContext } from '../../contexts/AlertContext';
+import { useAnalyticsContext } from '../../contexts/AnalyticsContext';
+import { AnalyticsEvent } from '../../lib/analytics';
+
+const promptConfig = {
+  title: 'Discard tag selection?',
+  description: 'Your personalized feed is only a few tags away',
+  cancelButton: {
+    title: 'Stay',
+    variant: ButtonVariant.Primary,
+    color: ButtonColor.Cabbage,
+  },
+  okButton: {
+    title: 'Leave',
+    variant: ButtonVariant.Secondary,
+  },
+  className: {
+    buttons: 'flex-row-reverse',
+  },
+};
+
+type OnboardingFeedHeaderProps = {
+  isPreviewFeedVisible: boolean;
+  setPreviewFeedVisible: React.Dispatch<React.SetStateAction<boolean>>;
+  isFeedPreviewEnabled: boolean;
+  tagsCount: number;
+};
+
+export const OnboardingFeedHeader = ({
+  isPreviewFeedVisible,
+  setPreviewFeedVisible,
+  isFeedPreviewEnabled,
+  tagsCount,
+}: OnboardingFeedHeaderProps): JSX.Element => {
+  let stopNav = true;
```

Comment:
This needs to go inside the useEffect right?

Reply:
That works too.